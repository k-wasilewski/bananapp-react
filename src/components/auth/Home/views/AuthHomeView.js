import React, {Component} from 'react';
import {Redirect} from "react-router-dom";
import {connect} from "react-redux";

class AuthHomeView extends Component {
    render() {
        if (this.props.prediction!==undefined && this.props.img!==undefined)
            return (<Redirect to={{
                pathname: '/auth/results',
                state: { prediction: this.props.prediction,
                    img: this.props.img}
            }}/>)
        else
            return (
            <div>
                <header>
                    <input type='file' name='avatar' onChange={this.props.fileChangedHandler} />
                    <button type='button' onClick={this.props.submitLoading} > Upload </button>
                    { this.props.loadingComponent }
                    { this.props.errorMsg }
                    { this.props.$imagePreview }
                </header>
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        username: state.setUsernameReducer.username,
    };
};

export default connect(mapStateToProps, null)(AuthHomeView);