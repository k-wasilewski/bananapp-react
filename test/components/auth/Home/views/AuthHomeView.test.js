import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../../../src/redux/store";
import {BrowserRouter} from "react-router-dom";
import AuthHomeView from "../../../../../src/components/auth/Home/views/AuthHomeView";
import {Provider} from "react-redux";

describe("AuthHomeView rendering specification", () => {
    it('AuthHomeView is rendered', () => {
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <AuthHomeView />
                </BrowserRouter>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});