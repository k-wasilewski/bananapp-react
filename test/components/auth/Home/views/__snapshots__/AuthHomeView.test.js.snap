// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHomeView rendering specification AuthHomeView is rendered 1`] = `
<div>
  <header>
    <input
      name="avatar"
      type="file"
    />
    <button
      type="button"
    >
       Upload 
    </button>
  </header>
</div>
`;
